### ISSUE TITLE
Document composables concept

### ISSUE THREAD

**Issue by pi0** [nuxt/nuxt#14723]
Nuxt 3 uses Vue 3 with Composition API and [unjs/unctx](https://github.com/unjs/unctx) for supporting composables with the implicit context in nuxtApp (plugins, middleware) and nuxt kit too.

However, there are many cases in this needs a better explanation of how it works internally to understand why an issue like [nuxt instance unavailable](https://github.com/nuxt/framework/issues?q=nuxt+instance+unavailable) happens and how to fix it with proper usage.

I've made an outline story in https://github.com/nuxt/nuxt.js/issues/14269 which can probably be used as a basis to initiate this concept page.

**Comment by floriankapaun** [comment]
> I was able to test callWithNuxt and it's working just fine. I've finally been able to beat multiple awaits and build Nuxt 2-like useAsyncData in my code, even though I had to use this function.

Was able to achieve a [workaround to my problem](https://stackoverflow.com/questions/76033123/nuxt-usefetch-useasyncdata-twice-in-method-throws-error-nuxt-instance-not-avai/76033170#76033170) with you help, but is that really the recommended approach to do things or is there a better way? 😬

**Comment by daniluk4000** [comment]
@floriankapaun I've created discussion to improve DX of this behavior with more helper functions, and also to document this callWithNuxt approach: https://github.com/nuxt/nuxt/discussions/19567

Sadly discussion received no response :(

I've worked much with Vue and Nuxt internals recently and I can say that callWithNuxt now is the only way to not lose context. In my opinion, if there is no coming fix for context from Vue team this needs to be properly documented and maybe DX-improved. 

Speaking of myself, I didn't find any better way to not lose app context on SSR but that.
